base/inference.jl

```diff
@@ -191,7 +193,7 @@ mutable struct InferenceState
                     vararg_type = rewrap(vararg_type, linfo.specTypes)
                 end
                 s_types[1][la] = VarState(vararg_type, false)
-                src.slottypes[la] = widenconst(vararg_type)
+                src.slottypes[la] = vararg_type
```

Comment:
The weird thing is that just a few lines above, we explicitly do

```
                if isa(atyp, DataType) && isdefined(atyp, :instance)
                    atyp = Const(atyp.instance)
```

and then call `widenconst(atyp)`! I'll try to find a test case.